force-app/main/default/classes/CON_ContactMerge_CTRL.cls

```diff
@@ -950,98 +950,137 @@ public with sharing class CON_ContactMerge_CTRL {
-                    // But instead when you pass a null directly to put, it works fine. 
-                    // And hence, this if statement.
-                    if (selectedRecord.get(row.fieldName) == null) {
-                        master.put(row.fieldName, null);
-                    } else {
-                        Object val = selectedRecord.get(row.fieldName);
-                        master.put(row.fieldName, val);
-                    }
-                }
+    private List<Contact> groupLosingContacts(Contact winningContact) {
+        List<Contact> losingerContacts = new List<Contact>();
+        for (Contact c : this.selectedRecords.values()) {
+            if (c.Id != winningContact.Id) {
+                losingerContacts.add(c);
             }
+        }
 
-            // Group all the loosing records separately.
-            List<Contact> losers = new List<Contact>();
+        return losingerContacts;
+    }
 
-            for (Contact c : this.selectedRecords.values()) {
-                if (c.id != master.id) {
-                    losers.add(c);
-                }
+    // Update the field values of master record based on the selected value for each field.
+    private void populateContactWithNonUniqueFields(Contact winningContact) {
+        for (FieldRow row : fieldRows) {
+            if (row.showRadio
+                && !row.fieldName.equals(MASTER_KEY)
+                && row.selectedValue != winningContact.Id
+                && !UTIL_Describe.isFieldUnique('Contact', row.fieldName))
+            {
+                SObject selectedRecord = this.selectedRecords.get(row.selectedValue);
+                Object val = selectedRecord.get(row.fieldName);
+                winningContact.put(row.fieldName, val);
             }
+        }
+    }
 
-            // before proceeding further first lock the records for change
-            List<Contact> allContacts = new List<Contact>();
-            allContacts.add((Contact)master);
-            allContacts.addAll(losers);
-            List<Contact> lock = [
-                SELECT Id 
-                FROM Contact 
-                WHERE Id IN :allContacts for update];
-
-            if (lock == null || lock.size() != allContacts.size()) {
-                ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.Error, 
-                    label.npe01.Contact_Merge_Error_Lock_failed  + ' ' + 
-                    label.npe01.Contact_Merge_Error_please_retry) );
-                return null;
+    private Contact populateContactWithOnlyUniqueFields(Id winningContactId) {
+        Contact winningContact = new Contact(Id = winningContactId);
+        for (FieldRow row : fieldRows) {
+            if (row.showRadio
+                && !row.fieldName.equals(MASTER_KEY)
+                && row.selectedValue != winningContactId
+                && UTIL_Describe.isFieldUnique('Contact', row.fieldName))
```

Comment:
For now, I think a Unit test that checks for our "test" unique field on Contact could be sufficient to cover this method. We can create a separate test or flow later.

Reply:
I add a whole regression flow to this